### PR TITLE
Support for comparison between variables

### PR DESCRIPTION
**Pull request by ariannab**
This request includes:

- Support for comparison between variables, like parameters. For example:
Comment: `if min > max `
Expected condition: `args[0]>args[1]`

- Fix of some goal files

- A little refactoring

### CODE REVIEW

File: src/main/java/org/toradocu/translator/ConditionTranslator.java [hunk 1]
```diff
@@ -106,51 +108,51 @@ private static String addPlaceholders(String text) {
...
+      inequalities.add(text.substring(matcherVarComp.start(), matcherVarComp.end()));
+      placeholderText = placeholderText.replaceFirst(INEQUALITY_VAR_REGEX, PLACEHOLDER_PREFIX + i);
+      placeholderText = findVerb(placeholderText, i);
       i++;
     }
 
     return placeholderText;
   }
 
+  private static String findVerb(String placeholderText, int i) {
```
Comment (albertogoffi) [inline src/main/java/org/toradocu/translator/ConditionTranslator.java]: Could you please provide the Javadoc documentation of this method?

File: src/test/java/org/toradocu/PrecisionRecallCommonsMath3.java [hunk 2]
```diff
@@ -47,7 +54,7 @@ public void testPrimes() throws Exception {
 
   @Test
   public void testBitsStreamGenerator() throws Exception {
-    test("org.apache.commons.math3.random.BitsStreamGenerator", 1, 0.5, 1, 0.8, 1, 1);
+    test("org.apache.commons.math3.random.BitsStreamGenerator", 0.5, 0.5, 1, 0.8, 1, 1);
```
Comment (albertogoffi) [inline src/test/java/org/toradocu/PrecisionRecallCommonsMath3.java]: Do we know why the precision in this case is worse than before?

File: src/test/java/org/toradocu/PrecisionRecallCommonsMath3.java [hunk 3]
```diff
@@ -22,7 +22,14 @@ public void testGaussian() throws Exception {
 
   @Test
   public void testUnivariateSolverUtils() throws Exception {
-    test("org.apache.commons.math3.analysis.solvers.UnivariateSolverUtils", 1, 0.777, 1, 0, 1, 1);
+    test(
+        "org.apache.commons.math3.analysis.solvers.UnivariateSolverUtils",
+        0.889,
```
Comment (albertogoffi) [inline src/test/java/org/toradocu/PrecisionRecallCommonsMath3.java]: Do we know why the precision in this case is worse than before?

File: src/main/java/org/toradocu/translator/ConditionTranslator.java [hunk 4]
```diff
@@ -314,6 +311,45 @@ private static void translatePropositions(
     }
   }
 
+  private static String findMatchingCodeElements(
```
Comment (albertogoffi) [inline src/main/java/org/toradocu/translator/ConditionTranslator.java]: Could you please provide the Javadoc documentation of this method?

File: src/test/java/org/toradocu/PrecisionRecallCommonsMath3.java (hunk 3 above)
Comment (ariannab) [inline src/test/java/org/toradocu/PrecisionRecallCommonsMath3.java]: Yes. This is due to the following condition:
Comment:` if lower >= initial or initial >= upper`.

Here we have a composition of conditions, currently not supported by Toradocu. So, before, this resulted in a totally empty condition. Now, the first part results in a translation, but the second cannot be translated: the whole result is a wrong condition.

File: src/test/java/org/toradocu/PrecisionRecallCommonsMath3.java (hunk 2 above)
Comment (ariannab) [inline src/test/java/org/toradocu/PrecisionRecallCommonsMath3.java]: Similar cases to the one above. Here we have 2 composite conditions, before empty, now partially translated - but not completely, thus, wrong condition.

File: src/main/java/org/toradocu/translator/ConditionTranslator.java (hunk 1 above)
Comment (ariannab) [inline src/main/java/org/toradocu/translator/ConditionTranslator.java]: Of course, I'm going to provide it.

**Pull request merged**